src/AppInstallerCommonCore/Manifest/ManifestSchemaValidation.cpp

```diff
@@ -99,82 +99,68 @@ namespace AppInstaller::Manifest::YamlParser
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_1_LOCALE), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            default:
-                THROW_HR(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED));
-            }
+            resourceMap = {
+                { ManifestTypeEnum::Singleton, IDX_MANIFEST_SCHEMA_V1_1_SINGLETON },
+                { ManifestTypeEnum::Version, IDX_MANIFEST_SCHEMA_V1_1_VERSION },
+                { ManifestTypeEnum::Installer, IDX_MANIFEST_SCHEMA_V1_1_INSTALLER },
+                { ManifestTypeEnum::DefaultLocale, IDX_MANIFEST_SCHEMA_V1_1_DEFAULTLOCALE },
+                { ManifestTypeEnum::Locale, IDX_MANIFEST_SCHEMA_V1_1_LOCALE },
+            };
         }
         else if (manifestVersion >= ManifestVer{ s_ManifestVersionV1 })
         {
-            switch (manifestType)
-            {
-            case AppInstaller::Manifest::ManifestTypeEnum::Singleton:
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_SINGLETON), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            case AppInstaller::Manifest::ManifestTypeEnum::Version:
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_VERSION), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            case AppInstaller::Manifest::ManifestTypeEnum::Installer:
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_INSTALLER), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            case AppInstaller::Manifest::ManifestTypeEnum::DefaultLocale:
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_DEFAULTLOCALE), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            case AppInstaller::Manifest::ManifestTypeEnum::Locale:
-                schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_V1_LOCALE), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
-                break;
-            default:
-                THROW_HR(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED));
-            }
+            resourceMap = {
+                { ManifestTypeEnum::Singleton, IDX_MANIFEST_SCHEMA_V1_SINGLETON },
+                { ManifestTypeEnum::Version, IDX_MANIFEST_SCHEMA_V1_VERSION },
+                { ManifestTypeEnum::Installer, IDX_MANIFEST_SCHEMA_V1_INSTALLER },
+                { ManifestTypeEnum::DefaultLocale, IDX_MANIFEST_SCHEMA_V1_DEFAULTLOCALE },
+                { ManifestTypeEnum::Locale, IDX_MANIFEST_SCHEMA_V1_LOCALE },
+            };
         }
         else
         {
-            schemaStr = JsonSchema::LoadResourceAsString(MAKEINTRESOURCE(IDX_MANIFEST_SCHEMA_PREVIEW), MAKEINTRESOURCE(MANIFESTSCHEMA_RESOURCE_TYPE));
+            idx = IDX_MANIFEST_SCHEMA_PREVIEW;

```

Comment:
>idx = IDX_MANIFEST_SCHEMA_PREVIEW; [](http://example.com/codeflow?start=12&length=34)

this should be:
resourceMap = {
                { ManifestTypeEnum::Preview, IDX_MANIFEST_SCHEMA_PREVIEW },
            };

to force the manifestType to be correct from caller